### PR TITLE
Magic Link support

### PR DESCRIPTION
**Pull request by komninoschat**

### CODE REVIEW

File: src/test/custom/emails/passwordless/html.ejs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+<body>
+  <h2>Hello <%= display_name %>
+  </h2>
+  <p>Please confirm your magic link <%= action %> by clicking the below button:</p>
+  <a href="<%= url %>/auth/magic-link?token=<%= token %>&action=<%= action.replace(/ /g, '-') %>">
+    <button>Confirm <%= action %></button></a>
+  <p>Thanks,<br />The Team</p>
+</body>
+
+</html>
```
Comment (elitan) [inline src/test/custom/emails/passwordless/html.ejs:32]: Use the same email template as the updated one on HBP.

File: src/Auth.test.ts [hunk 2]
```diff
@@ -15,6 +15,28 @@ it("should register second user", async () => {
...
+
+it("should not register a magic link user when magic link mode is disabled", async () => {
+  await nhost.withEnv({
+    ENABLE_MAGIC_LINK: 'false',
+  }, async () => {
+    await expect(
+      auth.register({ email: "magic-link-user@nhost.io" })
+    ).toReject();
+  })
+});
```
Comment (elitan) [inline src/Auth.test.ts:38]: Do we also want a test where the `password` is empty? 

Let's say the user should fill in a form to register with email and password, and only fill in `email` and click "Register". How will this be handled by the js sdk? Will the user "accidently" register using a magic link? PReferably the js sdk should throw an error.

Should work:

```js
const email = 'johan@nhost.io';
nhost.auth.register({email});
```

Should fail:
```js
const email = 'johan@nhost.io';
const password = '';
nhost.auth.register({email, password});
```

```js
const email = 'johan@nhost.io';
const password;
nhost.auth.register({email, password});
```

**Review by elitan — CHANGES_REQUESTED** [review]
